backend/search/query_test.go

```diff
@@ -22,18 +27,33 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+// https://github.com/bcc-code/bcc-media-platform/pull/1021
 func Test_ElasticQueryBasic(t *testing.T) {
-	if !elasticTestsEnabled {
-		t.Skip("Ealstic tests not enabled. Enable with -elastic flag")
-	}
+	rand.Seed(time.Now().UnixNano())
+	randomNumber := rand.Intn(1000000000000000)
+	testIndexName := fmt.Sprintf("bccm-integration-test-%d", randomNumber)
 
 	ctx := context.Background()
 	client := newElasticClient(ctx, ElasticConfig{
-		URL:      "http://localhost:9200/",
-		Username: "elastic",
-		Password: "bccm123",
+		CloudID: CloudId,
+		ApiKey:  ApiKey,
 	})
 
+	mapping := types.NewNestedProperty()
+	mapping.Properties = map[string]types.Property{
+		"total": types.NewIntegerNumberProperty(),
+		"free":  types.NewIntegerNumberProperty(),
+		"used":  types.NewIntegerNumberProperty(),
+	}
+
+	client.Indices.Create(testIndexName).Do(ctx)
+	client.Indices.PutMapping(testIndexName).Request(&putmapping.Request{Properties: map[string]types.Property{
```

Comment:
what does the index look like? Can you point me to the definition of the index?

Reply:
@KillerX

Reply:
The idea is to get something similar to what we have in BMM: https://github.com/bcc-code/bmm-api/blob/develop/BMM.Api.Infrastructure.Tests/ElasticSearch/SearchTrackTests.cs

- create an index for the test
- fill it with the test data that is needed for this test
- run actual test
- delete everything

Reply:
https://github.com/bcc-code/bcc-media-platform/blob/master/backend/search/fields.go#L36